core/src/main/java/org/web3j/protocol/http/HttpService.java

```diff
@@ -131,13 +133,26 @@ protected InputStream performIO(String request) throws IOException {
 
         RequestBody requestBody = RequestBody.create(JSON_MEDIA_TYPE, request);
         Headers headers = buildHeaders();
-
-        okhttp3.Request httpRequest = new okhttp3.Request.Builder()
-                .url(url)
-                .headers(headers)
-                .post(requestBody)
-                .build();
-
+        okhttp3.Request httpRequest;
+        String url = urls[counter];
+        try {
+            httpRequest = new okhttp3.Request.Builder()
+                    .url(url)
+                    .headers(headers)
+                    .post(requestBody)
+                    .build();
+        } catch (IllegalStateException e) {
+            url = next();
+            if (url != null) {
+                httpRequest = new okhttp3.Request.Builder()
```

Comment:
encapsulate in `private okhttp3.Request buildHttpRequest(...)` instead of copy pasta

Reply:
You mean moving the check implementation inside the `okhttp3.Rquest` class?

Reply:
No. I mean write this method in HttpService class so you don't duplicate code.